Add AMP compatibility to Slideshow shortcode

kienstra opened the pull request:
#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
* Adds AMP compatibility to the Slideshow shortcode
* Reuses the Slideshow block [render method](https://github.com/Automattic/jetpack/blob/34e9a2337a70ceec7a467a87bf04a83821a96397/extensions/blocks/slideshow/slideshow.php#L54)
* No change on non-AMP endpoints

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* Enhances an existing feature

#### Testing instructions:
1. Activate the [AMP plugin](https://wordpress.org/plugins/amp/)
2. In AMP Settings, select Transitional mode
3. Create a new post 
4. Add a Shortcode block
5. Enter this, with ids from your environment:
```html
[gallery ids="1691,766,760" type="slideshow"]
```
The attribute `type="slideshow"` is what triggers the code below.

6. Preview the front-end of the AMP URL:

![amp-button](https://user-images.githubusercontent.com/4063887/78599669-28a98280-7817-11ea-8da1-9cc9d29ea79e.png)

7. Expected: there's an `<amp-carousel>`, and it displays captions if the images have them:
![slideshow](https://user-images.githubusercontent.com/4063887/78598819-9c4a9000-7815-11ea-8c42-6d9059cefe9a.gif)

This won't look exactly like the non-AMP endpoint, but it should have the same images.

8. Preview the non-AMP URL:

![amp-button](https://user-images.githubusercontent.com/4063887/78599696-3959f880-7817-11ea-9674-1fd76b6e53ba.png)

9. Checkout the `master` branch
10. Preview the non-AMP URL in another tab
11. Expected: the non-AMP URL looks the same with this PR and without

#### Proposed changelog entry for your changes:
* Add AMP compatibility to Slideshow shortcode

kienstra committed:
For the Slideshow shortcode in AMP, use the render method from the Slideshow block

Also, add some test cases for the AMP output,
and enqueue the stylesheet for the Slideshow block.

kienstra committed:
Add a test case for there being no valid ID

This should help to ensure regressions,
to maybe prevent wp_list_pluck()
from throwing an error if there's no 'id' index.

kienstra commented on modules/shortcodes/slideshow.php:
```diff
@@ -182,6 +183,17 @@ public function shortcode_callback( $attr ) {
 
+		if ( class_exists( 'Jetpack_AMP_Support' ) && Jetpack_AMP_Support::is_amp_request() ) {
+			// Load the styles and use the rendering method from the Slideshow block.
+			Jetpack_Gutenberg::load_styles_as_required( 'slideshow' );
```
Will the `Jetpack_Gutenberg` class always be available?

jeherve replied:
`Jetpack_Gutenberg` will be available, yes. 👍

kienstra replied:
OK, nice!

jeherve commented on tests/php/modules/shortcodes/test-class.slideshow.php:
```diff
@@ -48,34 +57,72 @@ public function test_shortcodes_slideshow() {
+	 */
+	public function get_slideshow_shortcode_amp() {
+		return array(
+			'without_autostart_attribute' => array(
+				'[gallery type="slideshow" ids="' . $this->ids . ']',
+				'<div class="wp-block-jetpack-slideshow wp-amp-block wp-block-jetpack-slideshow__autoplay wp-block-jetpack-slideshow__autoplay-playing" id="wp-block-jetpack-slideshow__1"><div class="wp-block-jetpack-slideshow_container swiper-container"><amp-carousel width="800" height="600" layout="responsive" type="slides" data-next-button-aria-label="Next Slide" data-prev-button-aria-label="Previous Slide ...
+			),
+			'with_autostart_attribute'    => array(
+				'[gallery type="slideshow" ids="' . $this->ids . '" autostart="true"]',
+				'<div class="wp-block-jetpack-slideshow wp-amp-block wp-block-jetpack-slideshow__autoplay wp-block-jetpack-slideshow__autoplay-playing" id="wp-block-jetpack-slideshow__2"><div class="wp-block-jetpack-slideshow_container swiper-container"><amp-carousel width="800" height="600" layout="responsive" type="slides" data-next-button-aria-label="Next Slide" data-prev-button-aria-label="Previous Slide ...
+			),
```
Those 2 tests fail on WordPress.com, because the images aren't the ones you're expecting:
https://github.com/Automattic/jetpack/blob/a952f7254a18c8e10ad504102c168d5a9dfaa97c/tests/php/modules/shortcodes/test-class.slideshow.php#L6-L10

I think you may need to update those tests to be a less exact maybe, to take into account different images?

kienstra replied:
Hi Jeremy!
Ah, good point. Sorry for the delay.

The [commits below](https://github.com/Automattic/jetpack/compare/230427c8af274e8f2de95dbedf5ae1126d71ee01...6bc182685ea157c20b6095932f9684669e07bef4) interpolate the image URLs instead of hard-coding them. Also, they account for the `loading="lazy"` attribute if that's enabled.

Do the Automattic tests pass now? I don't see the `matticbot` icon in the builds below.

Also, the [failed Travis tests](https://travis-ci.org/github/Automattic/jetpack/jobs/676436204#L604) don't look to be related to this PR, but I could be wrong.

kienstra replied:
Update: it looks like the `matticbot` says 'Required review' now, but I'm not sure what that means about the tests.

jeherve replied:
> the failed Travis tests don't look to be related to this PR, but I could be wrong

I've rebased the PR to include some changes we had to make earlier today following some changes in Core (see #15494).

jeherve replied:
> Do the Automattic tests pass now?

I'm afraid not.

```
[test_shortcodes_slideshow_amp] test_shortcodes_slideshow_amp with data set "without_autostart_attribute"
[19:26:51][test_shortcodes_slideshow_amp with data set "without_autostart_attribute"] Failed asserting that two strings are equal.
[19:26:51][test_shortcodes_slideshow_amp with data set "without_autostart_attribute"] modules/shortcodes/test-class.slideshow.php:136
 
[19:26:51][test_shortcodes_slideshow_amp with data set "without_autostart_attribute"] '<div class="wp-block-jetpack-slideshow wp-amp-block wp-block-jetpack-slideshow__autoplay wp-block-jetpack-slideshow__autoplay-playing" id="wp-block-jetpack-slideshow__1"><div class="wp-block-jetpack-slideshow_container swiper-container"><amp-carousel width="800" height="600" layout="responsive" type="slides" data-next-button-aria-label="Next Slide" data-prev-button-aria-label="Previous Slide" controls loop autoplay delay=3000 id="wp-block-jetpack-slideshow__amp-carousel__1" on="slideChange:wp-block-jetpack-slideshow__amp-pagination__1.toggle(index=event.index, value=true)"><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/380428471_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/380428471_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/380428471_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/380428471_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/380428471_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/380428471_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/380428471_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/380428471_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/380428471_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/380428471_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/380428471_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/380428471_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/380428471_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/380428471_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div></amp-carousel><a aria-label="Pause Slideshow" class="wp-block-jetpack-slideshow_button-pause" role="button" on="tap:wp-block-jetpack-slideshow__amp-carousel__1.toggleAutoplay(toggleOn=false),wp-block-jetpack-slideshow__1.toggleClass(class=wp-block-jetpack-slideshow__autoplay-playing,force=false)"></a><a aria-label="Play Slideshow" class="wp-block-jetpack-slideshow_button-play" role="button" on="tap:wp-block-jetpack-slideshow__amp-carousel__1.toggleAutoplay(toggleOn=true),wp-block-jetpack-slideshow__1.toggleClass(class=wp-block-jetpack-slideshow__autoplay-playing,force=true)"></a><amp-selector id="wp-block-jetpack-slideshow__amp-pagination__1" class="wp-block-jetpack-slideshow_pagination swiper-pagination swiper-pagination-bullets amp-pagination" on="select:wp-block-jetpack-slideshow__amp-carousel__1.goToSlide(index=event.targetOption)" layout="container"><button option="0" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 1" selected></button><button option="1" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 2" ></button><button option="2" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 3" ></button><button option="3" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 4" ></button><button option="4" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 5" ></button><button option="5" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 6" ></button><button option="6" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 7" ></button><button option="7" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 8" ></button><button option="8" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 9" ></button><button option="9" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 10" ></button><button option="10" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 11" ></button><button option="11" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 12" ></button><button option="12" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 13" ></button><button option="13" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 14" ></button></amp-selector></div></div>'
[19:26:51][test_shortcodes_slideshow_amp with data set "without_autostart_attribute"] '<div class="wp-block-jetpack-slideshow wp-amp-block wp-block-jetpack-slideshow__autoplay wp-block-jetpack-slideshow__autoplay-playing" id="wp-block-jetpack-slideshow__1"><div class="wp-block-jetpack-slideshow_container swiper-container"><amp-carousel width="800" height="600" layout="responsive" type="slides" data-next-button-aria-label="Next Slide" data-prev-button-aria-label="Previous Slide" controls loop autoplay delay=3000 id="wp-block-jetpack-slideshow__amp-carousel__1" on="slideChange:wp-block-jetpack-slideshow__amp-pagination__1.toggle(index=event.index, value=true)"><div class="wp-block-jetpack-slideshow_slide"><figure><img width="800" height="600" src="http://example.org/files/2020/04/380428471_image1.jpg?w=800" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="800" height="600" src="http://example.org/files/2020/04/1643280503_image1.jpg?w=800" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div></amp-carousel><a aria-label="Pause Slideshow" class="wp-block-jetpack-slideshow_button-pause" role="button" on="tap:wp-block-jetpack-slideshow__amp-carousel__1.toggleAutoplay(toggleOn=false),wp-block-jetpack-slideshow__1.toggleClass(class=wp-block-jetpack-slideshow__autoplay-playing,force=false)"></a><a aria-label="Play Slideshow" class="wp-block-jetpack-slideshow_button-play" role="button" on="tap:wp-block-jetpack-slideshow__amp-carousel__1.toggleAutoplay(toggleOn=true),wp-block-jetpack-slideshow__1.toggleClass(class=wp-block-jetpack-slideshow__autoplay-playing,force=true)"></a><amp-selector id="wp-block-jetpack-slideshow__amp-pagination__1" class="wp-block-jetpack-slideshow_pagination swiper-pagination swiper-pagination-bullets amp-pagination" on="select:wp-block-jetpack-slideshow__amp-carousel__1.goToSlide(index=event.targetOption)" layout="container"><button option="0" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 1" selected></button><button option="1" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 2" ></button></amp-selector></div></div>'
[19:26:51][test_shortcodes_slideshow_amp] test_shortcodes_slideshow_amp with data set "with_autostart_attribute"
[19:26:51][test_shortcodes_slideshow_amp with data set "with_autostart_attribute"] Failed asserting that two strings are equal.
[19:26:51][test_shortcodes_slideshow_amp with data set "with_autostart_attribute"] modules/shortcodes/test-class.slideshow.php:136
 
[19:26:51][test_shortcodes_slideshow_amp with data set "with_autostart_attribute"] '<div class="wp-block-jetpack-slideshow wp-amp-block wp-block-jetpack-slideshow__autoplay wp-block-jetpack-slideshow__autoplay-playing" id="wp-block-jetpack-slideshow__2"><div class="wp-block-jetpack-slideshow_container swiper-container"><amp-carousel width="800" height="600" layout="responsive" type="slides" data-next-button-aria-label="Next Slide" data-prev-button-aria-label="Previous Slide" controls loop autoplay delay=3000 id="wp-block-jetpack-slideshow__amp-carousel__2" on="slideChange:wp-block-jetpack-slideshow__amp-pagination__2.toggle(index=event.index, value=true)"><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="1" height="1" src="http://example.org/files/2020/04/16651216_image1.jpg" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div></amp-carousel><a aria-label="Pause Slideshow" class="wp-block-jetpack-slideshow_button-pause" role="button" on="tap:wp-block-jetpack-slideshow__amp-carousel__2.toggleAutoplay(toggleOn=false),wp-block-jetpack-slideshow__2.toggleClass(class=wp-block-jetpack-slideshow__autoplay-playing,force=false)"></a><a aria-label="Play Slideshow" class="wp-block-jetpack-slideshow_button-play" role="button" on="tap:wp-block-jetpack-slideshow__amp-carousel__2.toggleAutoplay(toggleOn=true),wp-block-jetpack-slideshow__2.toggleClass(class=wp-block-jetpack-slideshow__autoplay-playing,force=true)"></a><amp-selector id="wp-block-jetpack-slideshow__amp-pagination__2" class="wp-block-jetpack-slideshow_pagination swiper-pagination swiper-pagination-bullets amp-pagination" on="select:wp-block-jetpack-slideshow__amp-carousel__2.goToSlide(index=event.targetOption)" layout="container"><button option="0" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 1" selected></button><button option="1" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 2" ></button><button option="2" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 3" ></button><button option="3" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 4" ></button><button option="4" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 5" ></button><button option="5" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 6" ></button><button option="6" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 7" ></button><button option="7" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 8" ></button><button option="8" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 9" ></button><button option="9" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 10" ></button><button option="10" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 11" ></button><button option="11" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 12" ></button><button option="12" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 13" ></button><button option="13" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 14" ></button><button option="14" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 15" ></button><button option="15" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 16" ></button></amp-selector></div></div>'
[19:26:51][test_shortcodes_slideshow_amp with data set "with_autostart_attribute"] '<div class="wp-block-jetpack-slideshow wp-amp-block wp-block-jetpack-slideshow__autoplay wp-block-jetpack-slideshow__autoplay-playing" id="wp-block-jetpack-slideshow__2"><div class="wp-block-jetpack-slideshow_container swiper-container"><amp-carousel width="800" height="600" layout="responsive" type="slides" data-next-button-aria-label="Next Slide" data-prev-button-aria-label="Previous Slide" controls loop autoplay delay=3000 id="wp-block-jetpack-slideshow__amp-carousel__2" on="slideChange:wp-block-jetpack-slideshow__amp-pagination__2.toggle(index=event.index, value=true)"><div class="wp-block-jetpack-slideshow_slide"><figure><img width="800" height="600" src="http://example.org/files/2020/04/16651216_image1.jpg?w=800" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div><div class="wp-block-jetpack-slideshow_slide"><figure><img width="800" height="600" src="http://example.org/files/2020/04/1489048507_image1.jpg?w=800" class="wp-block-jetpack-slideshow_image" alt="" object-fit="contain" /></figure></div></amp-carousel><a aria-label="Pause Slideshow" class="wp-block-jetpack-slideshow_button-pause" role="button" on="tap:wp-block-jetpack-slideshow__amp-carousel__2.toggleAutoplay(toggleOn=false),wp-block-jetpack-slideshow__2.toggleClass(class=wp-block-jetpack-slideshow__autoplay-playing,force=false)"></a><a aria-label="Play Slideshow" class="wp-block-jetpack-slideshow_button-play" role="button" on="tap:wp-block-jetpack-slideshow__amp-carousel__2.toggleAutoplay(toggleOn=true),wp-block-jetpack-slideshow__2.toggleClass(class=wp-block-jetpack-slideshow__autoplay-playing,force=true)"></a><amp-selector id="wp-block-jetpack-slideshow__amp-pagination__2" class="wp-block-jetpack-slideshow_pagination swiper-pagination swiper-pagination-bullets amp-pagination" on="select:wp-block-jetpack-slideshow__amp-carousel__2.goToSlide(index=event.targetOption)" layout="container"><button option="0" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 1" selected></button><button option="1" class="swiper-pagination-bullet" tabindex="0" role="button" aria-label="Go to slide 2" ></button></amp-selector></div></div>'

```

jeherve reviewed (changes requested):
This works well. Reusing the work from the Slideshow block is a smart take on this. I only have one remark about this.

kienstra committed:
Attempt to get WordPress.com test to pass

Instead of hard-coding the image URL
in the expected markup,
dynamically replace it.

kienstra committed:
Correct typo of {{url_1}} that should have been {{url_2}

kienstra committed:
Account for lazy-loading images

It looks like at least one failed Travis build
is from lazy-loaded images

MattGeri committed:
Fix unit tests
- Apply a consistent size and make width/height variable

jeherve committed:
Block classes: remove any null class values from the class array

jeherve committed:
Slideshow block: autoplay should always be a boolean

jeherve committed:
Slideshow shortcode: only pass autoplay param if necessary

Since the block is set to autoplay as soon as you pass an autoplay string parameter, regardless of its value, let's not pass that parameter unless necessary.

jeherve committed:
Simplify AMP tests

Let's not aim to validate the full slideshow markup. Instead, focus on pieces that indicate that the parameters are respected, and that the block is used.

jeherve reviewed (dismissed):
I've made a few changes to this PR as tests were still problematic on WordPress.com.

- I cleaned up the array producing CSS classes for Gutenberg blocks, as it was previously outputting empty spaces for classes that were set as `null`
- I reworked the autostart / autoplay conditions, as things work a bit differently between the block and the shortcode: the shortcode is set to autostart by default, while the block will not autoplay by default, but will autoplay as soon as you provide an autoplay parameter.
- I simplified the tests. I don't think we need to check for the entirety of the slideshow markup; what's interesting in my opinion is whether the AMP markup is being used, and if the parameters are respected.

Let me know what you think about it!

jeherve commented:
Hm, tests are failing on PHP 5.6 because of this:
https://github.com/sebastianbergmann/phpunit/pull/2778

I'll need to work around this.

jeherve committed:
Move check for invalid IDs to its own test.

See https://github.com/Automattic/jetpack/pull/15319#issuecomment-625339897

MattGeri commented:
> I've made a few changes to this PR as tests were still problematic on WordPress.com.
> 
> * I cleaned up the array producing CSS classes for Gutenberg blocks, as it was previously outputting empty spaces for classes that were set as `null`
> * I reworked the autostart / autoplay conditions, as things work a bit differently between the block and the shortcode: the shortcode is set to autostart by default, while the block will not autoplay by default, but will autoplay as soon as you provide an autoplay parameter.
> * I simplified the tests. I don't think we need to check for the entirety of the slideshow markup; what's interesting in my opinion is whether the AMP markup is being used, and if the parameters are respected.
> 
> Let me know what you think about it!

LGTM! 👍

jeherve commented:
r207455-wpcom
